### PR TITLE
Add test cases

### PR DESCRIPTION
**Pull request by R-o-y**

### CODE REVIEW

File: lAyeR/lAyeRTests/Vector3DTests.swift [hunk 1]
```diff
@@ -33,12 +38,63 @@ class Vector3DTests: XCTestCase {
...
+        XCTAssertEqual(v.projectionLength(on: zero), 0)
+    }
+    
+    func testProjectionOfZero() {
+        let zero = Vector3D(x: 0, y: 0, z: 0)
+        let v = Vector3D(x: 1, y: 3, z: 9)
+        XCTAssertEqual(zero.projectionLength(on: v), 0)
+    }
+    
+    func testProjectionLengthOnPerpendicularVector() {
```
Comment (billstark) [inline lAyeR/lAyeRTests/Vector3DTests.swift:90]: 这个和前面的`testDotProductWithPerpendicularVector`是一样的吧？

Comment (R-o-y) [reply lAyeR/lAyeRTests/Vector3DTests.swift:90]: resolved

Comment (billstark) [reply lAyeR/lAyeRTests/Vector3DTests.swift:90]: ？？？怎么resolve的啊

Comment (R-o-y) [reply lAyeR/lAyeRTests/Vector3DTests.swift:90]: see the most recent commit:
the first is for testing dot product
the second is now changed to test projection

Comment (billstark) [reply lAyeR/lAyeRTests/Vector3DTests.swift:90]: kkkk noticed

**Review by billstark — APPROVED** [review]

**Pull request merged**